apps/frontend/core-dapp/src/features/connect/ConnectButton.tsx

```diff
@@ -1,36 +1,38 @@
 import { observer } from 'mobx-react-lite'
-import styled from 'styled-components'
-import { Button } from 'prepo-ui'
-import { Trans } from '@lingui/macro'
+import { Button, ButtonProps } from 'prepo-ui'
 import { useRootStore } from '../../context/RootStoreProvider'
 
-const Wrapper = styled.div`
-  border-radius: ${({ theme }): string => theme.borderRadius.md};
-  display: inline-flex;
-`
+type Props = {
+  hideWhenConnected?: boolean
+} & ButtonProps
 
-const Flex = styled.div`
-  display: flex;
-`
+const ConnectButton: React.FC<Props> = ({ hideWhenConnected, ...buttonProps }) => {
+  const { config, web3Store } = useRootStore()
+  const { defaultNetwork } = config
+  const { connecting, connected, isNetworkSupported } = web3Store
 
-const ConnectButton: React.FC = () => {
-  const { web3Store } = useRootStore()
-  const { signerState } = web3Store
-  const account = signerState.address
-
-  const onClickLogin = (): void => {
+  const handleSwitchNetwork = (): void => {
+    if (!isNetworkSupported) {
+      web3Store.setNetwork(defaultNetwork)
+      return
+    }
     web3Store.connect()
   }
 
-  if (account) return null
+  const buttonText = isNetworkSupported
+    ? 'Connect Wallet'
+    : `Switch to ${defaultNetwork.displayName ?? defaultNetwork.chainName}`
+
+  // ignore whether network is supported (e.g. header connect button)
+  if (hideWhenConnected && connected) return null
```

Comment:
nit:
```suggestion
  if (connected && hideWhenConnected) return null
```
to keep same order as one below